tools/qc/fastq_utils/fastq_pre_barcodes.xml

```diff
@@ -0,0 +1,246 @@
+        <param label="UMI read"  name="umi_read" argument="--umi_read" type="text"   help="File in which UMI read can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param label="UMI offset"  name="umi_offset" argument="--umi_offset" type="integer"   help="Offset (integer)"/>
+        <param label="UMI Size"  name="umi_size" argument="--umi_size" type="integer"   help="Number of bases after the offset"/>
+        <param label="Cell Read"  name="Cell_read" argument="--Cell_read" type="text"   help="File in which Cell can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param label="Cell Offset"  name="Cell_offset" argument="--Cell_offset" type="integer"   help="Offset"/>
+        <param label="Cell Size"  name="Cell_size" argument="--Cell_size" type="integer"   help="Number of bases after the offset"/>
+        <param label="Sample Read"  name="sample_read" argument="--sample_read" type="text"   help="File in which sample barcode can be found, in this format: (read1|read2|index1|index2|index3)"/>
+        <param label="Sample Offset"  name="sample_offset" argument="--sample_offset" type="integer"   help="Offset"/>
+        <param label="Sample Size"  name="sample_size" argument="--sample_size" type="integer"   help="Number of bases after the offset"/>
+        <param label="read1 Offset"  name="read1_offset" argument="--read1_offset" type="integer"   help="None"/>
+        <param label="read1 Size"  name="read1_size" argument="--read1_size" type="integer"   help="None"/>
+        <param label="read2 Offset"  name="read2_offset" argument="--read2_offset" type="integer"   help="None"/>
+        <param label="read2 Size"  name="read2_size" argument="--read2_size" type="integer"   help="None"/>
+        <param label="Use 10x tags"  name="use_10x" argument="--10x" type="text"   help="Use 10X UMI tags (UB and UY) instead of the default tags defined in the SAM specification"/>
+        <param label="sam" name="sam" argument="--sam" type="text" help="No available description"/>
+    </inputs>
+    <outputs>
+        <data label="${tool.name} on ${on_string}: Output file 1" name="outfile1" format='?' />
+        <data label="${tool.name} on ${on_string}: Output file 2" name="outfile2" format='?' />
+        <data label="${tool.name} on ${on_string}: Output file 3" name="outfile3" format='?' />
+    </outputs>
+    <tests>
+        <test>
+            <param name="index1" value="test-data/barcode_test_1.fastq.gz"/>
+            <param name="phred_encoding" value="33"/>
+            <param name="min_qual" value="10"/>
+            <param name="umi_read" value="index1"/>
+            <param name="umi_offset" value="0"/>
+            <param name="umi_size" value="16"/>
+            <param name="read1_offset" value="0"/>
+            <param name="read1_size" value="-1"/>
+            <param name="read1" value="test-data/barcode_test_2.fastq.gz"/>
+            <output name="outfile1" file="test.fastq.gz"/>
```

Comment:
I'm not sure, but I suspect that you might need some assertion logic here. See galaxy tools docs and other tools' tests in this repo.

Reply:
That might explain why tests are skipped.

Reply:
Currently this is failing because diff won't compare files that are binary (gz in this case).